### PR TITLE
fix($injector): workaround for MS Edge class detection

### PR DESCRIPTION
**Pull request by lgalfaso**
Fix for MS Edge class detection

### CODE REVIEW

**Comment by lgalfaso** [comment]
@petebacondarwin can you take a look at this? Did some small so we do not have to keep track of what browser supports what ES6 feature

File: test/auto/injectorSpec.js [hunk 1]
```diff
@@ -247,30 +249,37 @@ describe('injector', function() {
...
+      if (support.fatArrow) {
         it('should take args before first arrow', function() {
           expect(annotate(eval('a => b => b'))).toEqual(['a']);
         });
+      }
 
+      if (support.classes) {
         it('should be possible to instantiate ES6 classes', function() {
           // Only Chrome (not even the FF we use) supports ES6 classes.
           if (!/chrome/i.test(navigator.userAgent)) return;
```
Comment (petebacondarwin) [inline test/auto/injectorSpec.js]: @lgalfaso should this chrome check be removed?

**Comment by petebacondarwin** [comment]
One question, otherwise LGTM

File: test/auto/injectorSpec.js (hunk 1 above)
Comment (lgalfaso) [inline test/auto/injectorSpec.js]: you are right

File: src/auto/injector.js [hunk 2]
```diff
@@ -826,8 +826,10 @@ function createInjector(modulesToLoad, strictDi) {
       if (msie <= 11) {
         return false;
       }
+      // Workaround for MS Edge.
+      // Check https://connect.microsoft.com/IE/Feedback/Details/2211653
       return typeof func === 'function'
-        && /^class\s/.test(Function.prototype.toString.call(func));
+        && /^(?:class\s|constructor\()/.test(Function.prototype.toString.call(func));
```
Comment (petebacondarwin) [inline src/auto/injector.js]: Actually I think we need to test for `constructor(` only if we are on Edge, since it is possible that someone might have a function actually called `constructor`.

Comment (petebacondarwin) [inline src/auto/injector.js]: @lgalfaso explained that:

```js
(function constructor() {}).toString() === "function constructor() {}"
```

So this is not an issue
